lib/Cloudflare.js

```diff
@@ -15,51 +15,75 @@ module.exports = (auth) => {
-  return function (url, size = 'normal', __callback = () => {}) {
-
+  return function (imageFile, __callback = () => {}) {
     // API Payload
-    let options = new FormData()
-    options.append('file', url)
+    try {
+      
+      const fData = new FormData()
+      fData.append('file', imageFile.buffer, { filename: imageFile.filename })
 
-    // Force timeout with 20s
-    let callbackSent = false
-    const callback = (err, data) => {
-      if (!callbackSent) {
-        callbackSent = true
-        __callback(err, data)
-      }
-      if (timer) {
-        clearTimeout(timer)
+      // Force timeout with 20s
+      let callbackSent = false
+      const callback = (err, data) => {
+        if (!callbackSent) {
+          callbackSent = true
+          __callback(err, data)
+        }
+        if (timer) {
+          clearTimeout(timer)
+        }
       }
-    }
 
-    // Verify if responded in 20s
-    const timer = setTimeout(() => {
-      callback(new Error('Cloudflare optimization timed out'))
-      logger.log(`Cloudflare timed out`)
-    }, 20000)
+      // Verify if responded in 30s
+      const timer = setTimeout(() => {
+        callback(new Error('Cloudflare optimization timed out'))
+        logger.log(`Cloudflare timed out`)
+      }, 30000)
+
+      // Upload image to cloudflare
+      cloudflareClient({
+        method: 'POST',
+        url: '/images/v1',
+        headers: { ...fData.getHeaders() },
+        data: fData
+      }).then((response) => {
+        // Retrieve Upload data
+        const id = response.data.result.id
+        const variants = response.data.result.variants
+        const fetchedImages = []
+        let processed = 0
 
-    // Upload image to cloudflare
-    cloudflareClient({
-      method: 'POST',
-      url: '/images/v1',
-      headers: { ...options.getHeaders() },
-      data: options
-    }).then((response) => {
-      const id = response.data.result.id
-      const variants = response.data.result.variants
-      const url = variants.find(v => v.endsWith(`/${size}`)) || variants[0]
-      if (url) {
-        download(url, { 'Accept': 'image/webp,image/*,*/*;q=0.8' }, (err, imageBody) => {
-          if (err) logger.error(err)
-          callback(err, {
-            ...response.data.result,
-            url: normalImage,
-            imageBody
-          })
-          setTimeout(() => cloudflareClient.delete(`/images/v1/${id}`), 60000)
+        // Download the correct variations
+        variants.forEach(vari => {
+          if (vari.includes('normal') || vari.includes('zoom') || vari.includes('big') ||  vari.includes('w90')) {
+            // Retrieve image type
+            const labels = vari.split('/')
+            const label = labels[labels.length - 1]
+            
+            // Download image, zoom must be jpeg
+            download(vari, { 'Accept': label === 'zoom' ? 'image/jpeg,image/*,*/*;q=0.8' : 'image/webp,image/*,*/*;q=0.8' }, (err, imageBody) => {
```

Comment:
```suggestion
            download(vari, { 'Accept': label === 'zoom' ? 'image/png,image/jpeg,image/*,*/*;q=1' : 'image/webp,image/*,*/*;q=0.8' }, (err, imageBody) => {
```

imagem `zoom` em qualidade 100%